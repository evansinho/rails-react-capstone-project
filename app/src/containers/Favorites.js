import React, { useState, useRef, useEffect } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as houseActions from '../actions/houseActions';
import Nav from '../components/Nav';
import Loader from '../components/Loader';
import CarouselCard from './CarouselCard';
import list from '../styles/List.module.css';
import fav from '../styles/Favorites.module.css';

const Favorites = ({ state, actions, history }) => {
  const [items, setItems] = useState(null);
  const [navOpen, setNavOpen] = useState(false);
  const parent = useRef(null);

  const checkLogged = () => {
    if (state.authenticated
      && (!localStorage.getItem('token') || localStorage.getItem('token') === 'undefined')) {
      localStorage.setItem('token', state.authenticated);
      localStorage.setItem('id', state.currentUser.id);
    } else if (localStorage.getItem('token') && !state.authenticated) {
      const token = localStorage.getItem('token');
      const currentU = localStorage.getItem('id');
      actions.fetchSubscription(`https://aqueous-wildwood-18424.herokuapp.com/users/${currentU}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      }, null, true);
    } else if (!state.authenticated) {
      history.push('/sign_in', state);
    } else if (state.authenticated
      && localStorage.getItem('token')
      && localStorage.getItem('token') !== 'undefined'
    ) {
      const token = localStorage.getItem('token');

      actions.fetchSubscription(`https://aqueous-wildwood-18424.herokuapp.com/user_meets/meets/${state.currentUser.id}/`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
    }
  };

  useEffect(checkLogged, [state.authenticated]);

  useEffect(() => {
    if (state.subscription) {
      setItems(state.subscription.res);
    }
  }, [state.subscription]);

  return (
    <main className={list['wrapper--favorites']} ref={parent}>
      <Nav open={navOpen} setNavOpen={setNavOpen} parent={parent} searchOpen={false} local="Favorites" />
      {
        items
          ? (
            <main className={fav.main}>
              {items.length && items.length > 0
                ? items.map(item => (
                  <CarouselCard key={item.id} id={`card-${item.id}`} item={item} name={item.name} type="product" history={history} state={state} />
                ))
                : (
                  <div className={list.title} style={{ flexDirection: 'column' }}>
                    <h1>You have not favorited any meet yet</h1>
                    <p><a href="/list">Check out the current meets here</a></p>
                  </div>
                )}
            </main>
          )
          : (<Loader />)
      }
    </main>
  );
};

function mapStateToProps({ state }) {
  return { state: { ...state } };
}

function mapActionsToProps(dispatch) {
  return {
    actions: bindActionCreators({ ...houseActions }, dispatch),
  };
}

Favorites.propTypes = {
  actions: PropTypes.objectOf(PropTypes.any).isRequired,
  history: PropTypes.objectOf(PropTypes.any).isRequired,
  state: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default connect(mapStateToProps, mapActionsToProps)(withRouter(Favorites));
