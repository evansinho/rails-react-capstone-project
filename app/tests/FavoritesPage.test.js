import React from 'react';
import renderer from 'react-test-renderer';
import Favorites from '../src/containers/Favorites';
import Wrapper from './helpers/Wrapper';

test('Detail is rendered', () => {
  const component = renderer.create(
    <Wrapper>
      <Favorites />
    </Wrapper>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
