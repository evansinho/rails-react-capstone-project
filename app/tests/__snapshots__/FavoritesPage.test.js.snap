// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail is rendered 1`] = `
<main
  className="wrapper--favorites"
>
  <nav
    className="main"
  >
    <ul>
      <li>
        <i
          aria-label="Open navigation menu"
          className="fas fa-bars"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={-1}
        />
      </li>
      <li>
        <a
          href="/list/"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="ico"
            data-icon="chevron-left"
            data-prefix="fal"
            focusable="true"
            role="button"
            viewBox="0 0 256 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              className=""
              d="M231.293 473.899l19.799-19.799c4.686-4.686 4.686-12.284 0-16.971L70.393 256 251.092 74.87c4.686-4.686 4.686-12.284 0-16.971L231.293 38.1c-4.686-4.686-12.284-4.686-16.971 0L4.908 247.515c-4.686 4.686-4.686 12.284 0 16.971L214.322 473.9c4.687 4.686 12.285 4.686 16.971-.001z"
              fill="currentColor"
            />
          </svg>
        </a>
        <h1
          className="title"
        >
          Favorites
        </h1>
      </li>
      <li>
        <input
          className="search-nav hidden"
          onKeyDown={[Function]}
          placeholder="Search..."
          type="text"
        />
        <i
          aria-label="Search"
          className="fas fa-search search-ico"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={-1}
        />
      </li>
    </ul>
  </nav>
  <div
    className="main"
  >
    <a
      className="main"
      href="/"
    >
      <svg
        className="circle"
        height="50"
        viewBox="0 0 100 100"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <linearGradient
            id="gradient"
            x1="0%"
            x2="0%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stopColor="#ec5200"
            />
            <stop
              offset="100%"
              stopColor="transparent"
            />
          </linearGradient>
        </defs>
        <circle
          cx="50"
          cy="50"
          fill="none"
          r="40"
          stroke="url(#gradient)"
          strokeWidth="13"
          transform="rotate(90 50 50)"
        />
      </svg>
      <div
        className="text-container"
      >
        <span
          className="text--light"
        >
          Circle
        </span>
      </div>
    </a>
  </div>
</main>
`;
